3515/2/commands/checkout.rb

```diff
@@ -0,0 +1,85 @@
+require_relative '../helper_modules/directory_maker'
+require_relative '../helper_modules/file_write'
+module CheckoutCommand
+  def checkout!(*)
+    if user_authorized?
+      checkin_true?
+    else
+      respond_with :message, text: "Извините, но ваш аккаунт не авторизован\nДля авторизации
+      введите команду /start"
+    end
+  end
+
+  def user_authorized?
+    !(User[from['id']]).to_s.empty?
+  end
+
+  def checkin_true?
+    if User[from['id']].checkin == 'true'
+      photo_request
+    else
+      respond_with :message, text: "Ошибка. Для начала нужно зачекиниться!\n
+      Введите команду /checkin"
+    end
+  end
+
+  def photo_request
+    respond_with :message, text: 'Отправь мне селфи :)'
+    save_context :check_api_photo_file_path
+  end
+
+  def check_api_photo_file_path(*)
+    if payload.include?('photo')
+      respond_with :message, text: 'Получил фото. Сохраняю...'
+      api_photo_file_true(payload)
+    else
+      respond_with :message, text: 'Ну я же вежливо попросил... '
+      photo_request
+    end
+  end
+
+  def api_photo_file_true(payload)
+    photo_id = payload['photo'].last['file_id']
+    uri = URI(ENV['API_URL'] + ENV['TOKEN'] +
+              '/getFile?file_id=' + photo_id)
+    photo_path = JSON.parse(Net::HTTP.get(uri))['result']['file_path']
+    photo_downloader(photo_path)
+  end
+
+  def photo_downloader(photo_path)
+    checkout_path = "public/#{from['id']}/checkout/#{Time.now}"
+    DirectoryMaker.dir_maker(checkout_path)
+    photo_path_new = "#{checkout_path}/selfie.jpg"
+    FileWrite.file_write(photo_path_new, URI(ENV['API_URL_FILE'] + ENV['TOKEN'] + '/' + photo_path))
+    request_geolocation
+  end
+
+  def request_geolocation
+    respond_with :message, text: 'А теперь отправь мне геолокацию...'
+    save_context :get_geolocation
+  end
+
+  def get_geolocation(*)
+    if payload.include?('location')
+      respond_with :message, text: 'Получил геолокацию. Проверяю...'
+      geolocation = payload['location'].values.to_a
+      geolocation_checker(geolocation)
+    # binding.pry
+    else
+      respond_with :message, text: 'Ну я же вежливо попросил...'
```

Comment:
The strings are also copy-pasted, so if you've got them on yamls, you don't need to change all their usages on the code, because the code just uses their keys and values are listed in a single place